ahcore/lit_module.py

```diff
@@ -56,9 +58,14 @@ def __init__(
                 "loss",
             ],
         )  # TODO: we should send the hyperparams to the logger elsewhere
-
-        self._num_classes = data_description.num_classes
-        self._model = model(out_channels=self._num_classes)
+        if isinstance(model, AhcoreJitModel):
+            self._model = model
+        elif isinstance(model, functools.partial):
```

Comment:
Before this PR, due to the monai dependency, the model is always partially initialised. Check the configs [here](https://github.com/NKI-AI/ahcore/tree/main/config/lit_module/monai_segmentation). With this PR, a new class of models are possible. 

But indeed, there should be another control flow to raise an error if the class of models doesn't match these two. However, I feel it's a bit constricted at the moment. As the repo matures, this will be clearer.